raiden-ts/src/path/epics.ts

```diff
@@ -466,6 +463,57 @@ export const pfsCapacityUpdateEpic = (
     ),
   );
 
+/**
+ * When monitoring a channel (either a new channel or a previously monitored one), send a matching
+ * PFSFeeUpdate to path_finding global room, so PFSs can pick us for mediation
+ * TODO: Currently, we always send Zero fees; we should send correct fee data from config
+ *
+ * @param action$ - Observable of channelMonitor actions
+ * @param state$ - Observable of RaidenStates
+ * @param deps - Raiden epic dependencies
+ * @returns Observable of messageGlobalSend actions
+ */
+export const pfsFeeUpdateEpic = (
+  action$: Observable<RaidenAction>,
+  state$: Observable<RaidenState>,
+  { log, address, network, signer, config$ }: RaidenEpicDeps,
+): Observable<messageGlobalSend> =>
+  action$.pipe(
+    filter(channelMonitor.is),
+    withLatestFrom(state$, config$),
+    // ignore actions while/if mediating not enabled
+    filter(([, , { pfsRoom, caps }]) => !!pfsRoom && !caps?.[Capabilities.NO_MEDIATE]),
+    mergeMap(([action, state, { pfsRoom }]) => {
+      const channel = state.channels[action.meta.tokenNetwork]?.[action.meta.partner];
+      if (channel?.state !== ChannelState.open) return EMPTY;
+
+      const message: PFSFeeUpdate = {
+        type: MessageType.PFS_FEE_UPDATE,
+        canonical_identifier: {
+          chain_identifier: bigNumberify(network.chainId) as UInt<32>,
+          token_network_address: action.meta.tokenNetwork,
+          channel_identifier: bigNumberify(channel.id) as UInt<32>,
+        },
+        updating_participant: address,
+        timestamp: new Date().toISOString().substr(0, 23) + '000',
```

Comment:
This looks rather hacky plus we already use a timestamp that PFS understands when dealing with IOUs. Probably makes sense to unify these two into one approach

Reply:
Unfortunately, this is not "what PFS understands", but instead, a very strict format which goes in the signature, so it needs to be exact. The signature expects a ISO date string with 26 characters, with microseconds resolution, and it does not accept the `Z` at the end of milliseconds which usually goes with JS Date to ISO string. So the `substr` is to ensure we "stop" at the milliseconds, and then we add zeroes to the microseconds. In the end, it's a string, and gets validated as string (see `packMessage`), but needs to be very specific, and I didn't find a standard way to do this on JS.

Reply:
I know, but probably the same as in https://github.com/raiden-network/light-client/blob/master/raiden-ts/src/path/epics.ts#L105 or do we need to have two separate ways of handling it?

Reply:
Hmm. As you can see below, python code does send with microseconds:
![image](https://user-images.githubusercontent.com/587021/79874981-671a7200-83bf-11ea-87ca-a33dd1a0afb7.png)
But I tested it by hand, and python's `PFSFeeUpdate` marshmallow can indeed deserialize with only the second resolution there, which is the same format as used in the signature and on the IOU, so I'm changing to standardize it there.
Thanks for the headsup